rfd/0169-auto-updates-linux-agents.md

```diff
@@ -91,45 +89,145 @@ spec:
   # cluster level. Only turn agent automatic updates off if self-managed
   # agent updates are in place.
   agent_auto_update: true|false
-  # agent_update_hour sets the hour in UTC at which clients should update their agents.
-  agent_update_hour: 0-23
-  # agent_update_jitter_seconds sets a duration in which the upgrade will occur after the hour.
-  # The agent upgrader will pick a random time within this duration to wait to upgrade.
-  agent_update_jitter_seconds: 0-3600
-  
-  [...]
+
+  # agent_auto_update_groups contains both "regular" and "critical" schedules.
+  # The schedule used is determined by the agent_version_schedule associated
+  # with the version in autoupdate_version.
+  # Groups are not configurable with the "immediate" schedule.
+  agent_auto_update_groups:
+    # schedule is "regular" or "critical"
+    regular:
+    - name: staging-group
+      # agent_selection defines which agents are included in the group.
+      agent_selection:
+        # query selects agents by resource query.
+        #  default: all connected agents
+        query: 'labels["environment"]=="staging"'
+      # days specifies the days of the week when the group may be upgraded.
+      #  default: ["*"] (all days)
+      days: [“Sun”, “Mon”, ... | "*"]
+      # start_hour specifies the hour when the group may start upgrading.
+      #  default: 0
+      start_hour: 0-23
+      # jitter_seconds specifies a maximum jitter duration after the start hour.
+      # The agent upgrader client will pick a random time within this duration to wait to upgrade.
+      #  default: 0
+      jitter_seconds: 0-60
+      # max_in_flight specifies the maximum number of agents that may be upgraded at the same time.
+      #  default: 100%
+      max_in_flight: 0-100%
```

Comment:
How do proxies decide which host IDs to instruct to upgrade, especially given the additional filtering?

Reply:
Mentioned here: https://github.com/gravitational/teleport/pull/40190/files#diff-4fb277bd3d884c7b045e7efad61111ac4e7957c063e0c0c9281f4e74fd72997fR61-R64

The `max_in_flight` hosts are selected randomly out of the total number of hosts in the group.

I'll clarify above that the selection is random.